### PR TITLE
Fix the loader issues

### PR DESCRIPTION
**Pull request by qburst-aparna**
Added fixes into the config.js to solve the babel loader issues while installing package into a consumer app.

Added the missing dependencies in package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
         "@atlaskit/tree": "^4.1.11",
         "@atlaskit/width-detector": "^0.3.1",
         "@babel/cli": "^7.2.3",
-        "@babel/core": "^7.3.4",
+        "@babel/core": "^7.0.0-0",
```
Comment (dpisani-atl) [inline package.json]: is this a downgrade? Why is this necessary?

File: package.json [hunk 2]
```diff
@@ -68,14 +68,15 @@
         "@emotion/core": "^10.0.9",
         "@emotion/styled": "^10.0.9",
         "@mdx-js/loader": "^0.17.5",
-        "@mdx-js/mdx": "^0.17.5",
+        "@mdx-js/mdx": "^0.15.0",
```
Comment (dpisani-atl) [inline package.json]: Same question here, why did we need to downgrade this package?

Comment (Noviny) [reply package.json]: Next had some peerdep mismatches if we use later versions, and we wanted to avoid adding errors on installing our stuff.

File: package.json (hunk 1 above)
Comment (qburst-aparna) [reply package.json]: the other babel plugins like `plugin-proposal-class-properties, plugin-proposal-object-rest-spread, preset-typescript` etc have dependency to this babel/core version and it was throwing error while installing our package.

File: package.json (hunk 2 above)
Comment (qburst-aparna) [reply package.json]: yes , `zeit/next-mdx` has dependency to the downgraded version
